datafusion/core/src/datasource/listing/table.rs

```diff
@@ -616,7 +582,7 @@ pub struct ListingTable {
     table_schema: SchemaRef,
     options: ListingOptions,
     definition: Option<String>,
-    collected_statistics: StatisticsCache,
+    collected_statistics: Option<FileStaticCache>,
```

Comment:
As written, I think this may regress performance for some users unexpectedly. Specifically, by default there will be no caching within the execution of a single query (which I think was the original usecase for this feature)

Thus I recommend this be

```suggestion
    collected_statistics: FileStaticCache,
```

And then rather `ListingTable::try_new_with_cache` add a function like

```rust
impl ListingTable {
    pub fn with_cache(mut self, cache: FileStaticCache) {
        self.collected_statistics = cache;
        self
}
```